File: src/main/java/no/nav/pensjon/regler/domain/beregning2011/AfpLivsvarig.java
package no.nav.pensjon.regler.domain.beregning2011;

import no.nav.pensjon.regler.domain.beregning.Ytelseskomponent;
import no.nav.pensjon.regler.domain.kode.FormelKodeCti;
import no.nav.pensjon.regler.domain.kode.YtelsekomponentTypeCti;

public class AfpLivsvarig extends Ytelseskomponent {
    private static final long serialVersionUID = 8482215632698056210L;

    private int justeringsbelop;
    private double afpProsentgrad;
    private double afpForholdstall;

    public double getAfpForholdstall() {
        return afpForholdstall;
    }

    public void setAfpForholdstall(double afpForholdstall) {
        this.afpForholdstall = afpForholdstall;
    }

    public double getAfpProsentgrad() {
        return afpProsentgrad;
    }

    public void setAfpProsentgrad(double afpProsentgrad) {
        this.afpProsentgrad = afpProsentgrad;
    }

    public AfpLivsvarig() {
        super();
        setFormelKode(new FormelKodeCti("AFPx"));
        ytelsekomponentType = new YtelsekomponentTypeCti("AFP_LIVSVARIG");
    }

    public AfpLivsvarig(AfpLivsvarig o) {
        super(o);
        afpForholdstall = o.afpForholdstall;
        afpProsentgrad = o.afpProsentgrad;
        justeringsbelop = o.justeringsbelop;
    }

    public int getJusteringsbelop() {
        return justeringsbelop;
    }

    public void setJusteringsbelop(int justeringsbelop) {
        this.justeringsbelop = justeringsbelop;
    }

}

File: src/main/kotlin/no/nav/pensjon/regler/domain/beregning2011/AfpLivsvarig.kt
package no.nav.pensjon.regler.domain.beregning2011

import com.fasterxml.jackson.annotation.JsonSubTypes
import com.fasterxml.jackson.annotation.JsonTypeInfo
import no.nav.pensjon.regler.domain.beregning.Ytelseskomponent
import no.nav.pensjon.regler.domain.enum.FormelKodeEnum
import no.nav.pensjon.regler.domain.enum.YtelseskomponentTypeEnum

@JsonSubTypes(
    JsonSubTypes.Type(value = FremskrevetAfpLivsvarig::class)
)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
open class AfpLivsvarig : Ytelseskomponent {
    var justeringsbelop = 0
    var afpProsentgrad = 0.0
    var afpForholdstall = 0.0

    override var ytelsekomponentTypeEnum: YtelseskomponentTypeEnum = YtelseskomponentTypeEnum.AFP_LIVSVARIG

    constructor() {
        formelKodeEnum = FormelKodeEnum.AFPx
    }

    constructor(o: AfpLivsvarig) : super(o) {
        afpForholdstall = o.afpForholdstall
        afpProsentgrad = o.afpProsentgrad
        justeringsbelop = o.justeringsbelop
    }
}